pkg/util/log/gophercloud.go

```diff
@@ -0,0 +1,73 @@
+package log
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+)
+
+func NewLoggingProviderClient(endpoint string, logger kitlog.Logger) (*gophercloud.ProviderClient, error) {
+	providerClient, err := openstack.NewClient(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	providerClient.HTTPClient.Transport = &loggingRoundTripper{
+		&http.Transport{},
+		logger,
+	}
+
+	return providerClient, err
+}
+
+type loggingRoundTripper struct {
+	rt     http.RoundTripper
+	Logger kitlog.Logger
+}
+
+func (lrt *loggingRoundTripper) RoundTrip(request *http.Request) (response *http.Response, err error) {
+	logger := kitlog.With(lrt.Logger,
```

Comment:
Looking at the implementation of `With` it doesn't come for free. There is an awful lot of shuffling around and allocation going on.
I don't think its pretty inefficient to prepare a single log line like this and we should try to use the contextual loggers only if they are used for a longer period and multiple log line.

Reply:
Optimized in 1393ff5